public_html/script/src/drawable/drawable.js

```diff
@@ -56,19 +69,31 @@ export const Drawable = class {
         this.sortChildren();
     }
 
-    removeFromParent() {
-        if (this[PARENT] !== undefined) {
-            this[PARENT].removeChild(this);
-        }
+    [ANIM_COMPLETE]() {
+        this[ANIM_COMPLETION]();
+        this[ANIM_COMPLETION] = undefined;
+        this[MOVEMENT_VECTOR] = new Point(0, 0);
     }
 
-    addToCanvas(canvasID) {
-        this[CANVAS_ID] = canvasID;
-        //this.draw(); TODO do we need this or should we keep it out?
+    animateToPoint({pt, speed, completion}) { //speed in pixels / frame
```

Comment:
What's wrong with passing a completion function? And how exactly would I promisify it?